lib/seadsa/DsaBottomUp.cc

```diff
@@ -93,26 +94,25 @@ void BottomUpAnalysis::cloneAndResolveArguments(const DsaCallSite &CS,
     const Value *onlyAllocSite = findUniqueReturnValue(callee);
     if (onlyAllocSite && !calleeG.hasAllocSiteForValue(*onlyAllocSite))
       onlyAllocSite = nullptr;
-    if (!flowSensitiveOpt)
-      onlyAllocSite = nullptr;
+    if (!flowSensitiveOpt) onlyAllocSite = nullptr;
 
     const Cell &ret = calleeG.getRetCell(callee);
     Node &n = C.clone(*ret.getNode(), false, onlyAllocSite);
     Cell c(n, ret.getRawOffset());
     nc.unify(c);
   }
 
-  // clone and unify actuals and formals
   DsaCallSite::const_actual_iterator AI = CS.actual_begin(),
                                      AE = CS.actual_end();
-  for (DsaCallSite::const_formal_iterator FI = CS.formal_begin(),
-                                          FE = CS.formal_end();
+  for (DsaCallSite::const_formal_iterator FI = CS.formal_func_begin(callee),
```

Comment:
Does not have `CS` already the callee, why do we need to pass it?

Reply:
we pass in the spec function if we have it, otherwise we use the callee that CS has (see line 86)

Reply:
we can't use the formals from the CS callee because `calleeG.hasCell(*fml)` would always fail if calleeG was generated using a spec function